src/components/Dashboard/SafeApps/Grid.tsx

```diff
@@ -35,26 +36,21 @@ const StyledLink = styled(Link)`
   }
 `
 
-const StyledIcon = styled.img`
-  width: 160px;
-  height: auto;
-`
-
 const MAX_APPS = 3
 
-const OFFICIAL_APPS: Record<string, string> = {
-  TRANSACTION_BUILDER: '29',
-  WALLET_CONNECT: '11',
-}
-
 const Grid = (): ReactElement => {
   const { allApps, pinnedSafeApps, togglePin } = useAppList()
-  const officialApps = allApps.filter((app) => Object.values(OFFICIAL_APPS).includes(app.id))
+  // Transactions Builder && Wallet connect
+  const officialAppIds = ['29', '11']
+  const pinnedSafeAppsIds = pinnedSafeApps.map((app) => app.id)
+  const officialApps = allApps.filter((app) => officialAppIds.includes(app.id) && !pinnedSafeAppsIds.includes(app.id))
 
   const displayedApps = pinnedSafeApps.concat(officialApps).slice(0, MAX_APPS)
```

Comment:
We could memoize this variable based on `pinnedSafeAppsIds` and `allApps`.

Reply:
Good one! I've memoized this data when `allApps` and `pinnedSafeApps` don't change